cmake/setup_finalize.cmake

```diff
@@ -55,12 +55,64 @@ endforeach()
 #
 
 foreach(build ${DEAL_II_BUILD_TYPES})
-  check_compiler_setup(
-    "${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_${build}}"
-    "${DEAL_II_LINKER_FLAGS} ${DEAL_II_LINKER_FLAGS_${build}}"
-    DEAL_II_HAVE_USABLE_FLAGS_${build}
-    ${DEAL_II_LIBRARIES} ${DEAL_II_LIBRARIES_${build}}
-    )
+  macro(_check_linker_flags)
+    check_compiler_setup(
+      "${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_${build}}"
+      "${DEAL_II_LINKER_FLAGS} ${DEAL_II_LINKER_FLAGS_${build}}"
+      DEAL_II_HAVE_USABLE_FLAGS_${build}
+      ${DEAL_II_LIBRARIES} ${DEAL_II_LIBRARIES_${build}}
+      )
+  endmacro()
+
+  macro(_drop_linker_flag _linker_flag _replacement_flag _variable)
+    message(STATUS
+      "Unable to compile a simple test program. "
+      "Trying to drop \"${_linker_flag}\" from the linker flags."
+      )
+    foreach(_flags
+        DEAL_II_LINKER_FLAGS DEAL_II_LINKER_FLAGS_${build}
+        BASE_LINKER_FLAGS BASE_LINKER_FLAGS_${build}
+        )
+      string(REPLACE "${_linker_flag}" "${_replacement_flag}"
+        ${_flags} "${${_flags}}"
+        )
+    endforeach()
+    set(${_variable} FALSE CACHE INTERNAL "" FORCE)
+    set(${_variable} FALSE)
```

Comment:
Would you mind refactoring the `set(${_variable ...)` part into its own macro?
It's a bit odd to put this into the logic above.